packages/hardhat-core/src/internal/cli/vars.ts

```diff
@@ -0,0 +1,223 @@
+    await getTaskDefinitionAndTaskArguments(allUnparsedCLAs);
+
+  switch (taskDefinition.name) {
+    case "set":
+      return set(taskArguments.key, taskArguments.value);
+    case "get":
+      return get(taskArguments.key);
+    case "list":
+      return list();
+    case "delete":
+      return del(taskArguments.key);
+    case "path":
+      return path();
+    case "setup":
+      return setup(configPath);
+    default:
+      return 1; // Error code
+  }
+}
+
+async function set(key: string, value?: string): Promise<number> {
+  const varsManager = HardhatContext.getHardhatContext().varsManager;
+
+  varsManager.validateKey(key);
+
+  varsManager.set(key, value ?? (await getVarValue()));
+
+  console.warn(
+    `Key-value pair stored at the following path: ${HardhatContext.getHardhatContext().varsManager.getStoragePath()}`
+  );
+
+  return 0;
+}
+
+function get(key: string): number {
+  const value = HardhatContext.getHardhatContext().varsManager.get(key);
+
+  if (value !== undefined) {
+    console.log(value);
+    return 0;
+  }
+
+  console.warn(
+    chalk.yellow(`There is no value associated to the key '${key}'`)
+  );
+  return 1;
+}
+
+function list(): number {
+  const keys = HardhatContext.getHardhatContext().varsManager.list();
+
+  if (keys.length > 0) {
+    keys.forEach((k) => console.log(k));
+
+    console.warn(
+      `\nAll the key-value pairs are stored at the following path: ${HardhatContext.getHardhatContext().varsManager.getStoragePath()}`
+    );
+  } else {
+    console.warn(chalk.yellow(`There are no key-value pairs stored`));
+  }
+
+  return 0;
+}
+
+function del(key: string): number {
+  if (HardhatContext.getHardhatContext().varsManager.delete(key)) {
+    console.warn(
+      `The key was deleted at the following path: ${HardhatContext.getHardhatContext().varsManager.getStoragePath()}`
+    );
+    return 0;
+  }
+
+  console.warn(
+    chalk.yellow(`There is no value associated to the key '${key}'`)
+  );
+
+  return 1;
+}
+
+function path() {
+  console.log(HardhatContext.getHardhatContext().varsManager.getStoragePath());
+  return 0;
+}
+
+function setup(configPath: string | undefined) {
+  log("Switching to VarsManagerSetup to collect vars");
+  HardhatContext.getHardhatContext().switchToSetupVarsManager();
+
+  try {
+    log("Loading config and tasks to trigger vars collection");
+    loadConfigFile(configPath);
+  } catch (err: any) {
+    console.error(
+      chalk.red(
+        "There is an error in your hardhat configuration file. Please double check it.\n"
+      )
+    );
+
+    // eslint-disable-next-line @nomicfoundation/hardhat-internal-rules/only-hardhat-error
+    throw err;
+  }
+
+  listVarsToSetup();
+
+  return 0;
+}
+
+function loadConfigFile(configPath: string | undefined) {
+  const configEnv = require(`../core/config/config-env`);
+
+  const globalAsAny: any = global;
+  Object.entries(configEnv).forEach(
+    ([key, value]) => (globalAsAny[key] = value)
+  );
```

Comment:
what is this doing?

Reply:
Loading all the global variables/functions/etc which are defined inside the config-env.ts file.
So you can access them inside your configuration file

Reply:
Got it! Can we add a small comment above to clarify it?